### PR TITLE
fix(core/gossip): Replace hardcoded string of vendors and trainers with gossip menu item

### PR DESCRIPTION
**Pull request by Bogir**
## Changes Proposed:
-  GOSSIP_TEXT_BROWSE_GOODS and GOSSIP_TEXT_TRAIN replaced wifh gossip menu items
-  Added condition to see menu items from GM-mode

## Issues Addressed:

- Closes some localisation issues

## Tests Performed:

- Tested on AzerothCore rev. 1f90a18a5888

## How to Test the Changes:

1. `.go c 97713`
In GM-mode:

изображение

If GM-mode is OFF:

изображение

2. `.go c 42201` and other trainers or vendors

In GM-mode

изображение

if GM-mode is OFF:

изображение

If patch is incorrect - you'll not see "Train me!" or "I'd like to browse your goods." 
If correct - you'll see them or your localized strings.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nefertumm** [comment]
Hi! Thanks for your PR! Could you change the Leatherworking part? It is addressed here https://github.com/azerothcore/azerothcore-wotlk/pull/10415 :D

**Review by Nefertumm — APPROVED** [review]

File: src/server/scripts/Kalimdor/zone_winterspring.cpp [hunk 1]
```diff
@@ -59,8 +59,8 @@ class npc_rivern_frostwind : public CreatureScript
         if (creature->IsQuestGiver())
             player->PrepareQuestMenu(creature->GetGUID());
 
-        if (creature->IsVendor() && player->GetReputationRank(589) == REP_EXALTED)
-            AddGossipItemFor(player, GOSSIP_ICON_VENDOR, GOSSIP_TEXT_BROWSE_GOODS, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_TRADE);
+        if (creature->IsVendor() && (player->GetReputationRank(589) == REP_EXALTED || player->IsGameMaster()))
```
Comment (Kitzunu) [inline src/server/scripts/Kalimdor/zone_winterspring.cpp:62]: Why adding a check for gamemaster

Comment (Bogir) [reply src/server/scripts/Kalimdor/zone_winterspring.cpp:62]: > Why adding a check for gamemaster

Only ещ give the opportunity to test in GЬ-mode, without fulfilling the all game conditions.

**Comment by Bogir** [comment]
Shall I continue this way?

File: src/server/game/AI/ScriptedAI/ScriptedGossip.h [hunk 2]
```diff
@@ -100,4 +97,47 @@ void SendGossipMenuFor(Player* player, uint32 npcTextID, Creature const* creatur
...
+constexpr std::pair<uint32, uint32> gossip_want_fly_green_flight    = { 9573, 1 };    // I want to fly on the wings of the green flight.
+constexpr std::pair<uint32, uint32> gossip_want_fly_bronze_flight   = { 9574, 1 };    // I want to fly on the wings of the bronze flight.
+constexpr std::pair<uint32, uint32> gossip_want_fly_red_flight      = { 9575, 0 };    // I want to fly on the wings of the red flight.
+
+constexpr std::pair<uint32, uint32> gossip_what_abil_emerald_have   = { 9573, 4 };    // What abilities do emerald drakes have?
+constexpr std::pair<uint32, uint32> gossip_what_abil_amber_have     = { 9574, 4 };    // What abilities do amber drakes have?
+constexpr std::pair<uint32, uint32> gossip_what_abil_ruby_have      = { 9575, 3 };    // What abilities do ruby drakes have?
+
+constexpr std::pair<uint32, uint32> gossip_where_do_we_go           = { 9708, 0 };    // So where do we go from here?
+
```
Comment (Nefertumm) [inline src/server/game/AI/ScriptedAI/ScriptedGossip.h:142]: No, each variable should be defined in the proper file, like an enum.

File: src/server/game/AI/ScriptedAI/ScriptedGossip.h [hunk 3]
```diff
@@ -92,6 +88,7 @@ void AddGossipItemFor(Player* player, uint32 icon, std::string const& text, uint
 
 // Uses gossip item info from DB
 void AddGossipItemFor(Player* player, uint32 gossipMenuID, uint32 gossipMenuItemID, uint32 sender, uint32 action, uint32 boxMoney = 0);
+void AddGossipItemFor(Player* player, std::pair<uint32, uint32> gossip, uint32 sender, uint32 action, uint32 boxMoney = 0);
```
Comment (Nefertumm) [inline src/server/game/AI/ScriptedAI/ScriptedGossip.h:91]: I would use a typedef with some significative name to define the pair.

**Comment by Nefertumm** [comment]
Merge conflicts, and the leatherworking ones are already fixed on master with 4806769beb160a13d35aef98c095e73b750b17be, so please don't change them.

**Comment by Bogir** [comment]
need rework :)

**Pull request closed without merging**